src/test/util_string_tests.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/string.h>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace util;
+
+BOOST_AUTO_TEST_SUITE(util_string_tests)
+
+BOOST_AUTO_TEST_CASE(ConstevalFormatString_NumSpec)
+{
+    // Compile-time sanity checks
+    static_assert([] {
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("%%");
+        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%s");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("%%s");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("s%%");
+        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%%%s");
+        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%s%%");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers(" 1$s");
```

Comment:
Tried adding
```
        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%02d");
```
here and it fails.

Ran
`git grep -E '"[^"]*%[0-9]+[aAcdfeEfFgGinopsuxX*][^"]*' src/ ':(exclude)src/leveldb' ':(exclude)src/secp256k1' ':(exclude)*.ts'`
and it feels like a common enough case to want to have fixed width formatting (*optionally* `0`-padded). Examples:
```
src/netbase.cpp:            LogError("Proxy requested wrong authentication method %02x\n", pchRet1[1]);
src/flatfile.cpp:                LogDebug(BCLog::VALIDATION, "Pre-allocating up to position 0x%x in %s%05u.dat\n", new_size, m_prefix, pos.nFile);
src/init.cpp:                LogPrintf(" block index %15dms\n", Ticks<std::chrono::milliseconds>(SteadyClock::now() - load_block_index_start_time));
src/node/blockstorage.cpp:            LogDebug(BCLog::BLOCKSTORAGE, "Prune: %s deleted blk/rev (%05u)\n", __func__, *it);
src/node/blockstorage.cpp:                          "Failed to flush previous block file %05i (finalize=1, finalize_undo=%i) before opening new block file %05i\n",
src/node/blockstorage.cpp:                LogPrintLevel(BCLog::BLOCKSTORAGE, BCLog::Level::Warning, "Failed to flush undo file %05i\n", _pos.nFile);
src/node/blockstorage.cpp:            LogPrintf("Reindexing block file blk%05u.dat...\n", (unsigned int)nFile);
```

Sorry for not realizing and raising this in an earlier review.